### ISSUE TITLE
Startup Applications don't seem to handle optional arguments correctly

### ISSUE THREAD

**Issue by ManIVIctorious** [linuxmint/Cinnamon#3988]
Some days ago i switched to geary as my preferred email-application. You can start geary with the option: 
`geary --hidden` (which does what it says -> no window pops up)
This works for the alt-f2 launcher and in terminals, i have added exactly this command to startup applications and every time i start my pc the geary window pops up.

When i have some time i will try with other applications too

**Comment by mtwebster** [comment]
Hi, what happens if you add a few second delay to its startup?

**Comment by ManIVIctorious** [comment]
Well, i have already tried that, a delay of 5 seconds doesn't seem to change anything. I can try with more if you want, but this would only be a workaround and no solution for the problem

**Comment by ManIVIctorious** [comment]
Oh, just if this is important:
`$ pacman -Qs cinnamon
local/cinnamon 2.4.8-1
    Linux desktop which provides advanced innovative features and a traditional user
    experience
local/cinnamon-control-center 2.4.2-3
    The Control Center for Cinnamon
local/cinnamon-desktop 2.4.2-1
    Library with common API for various Cinnamon modules
local/cinnamon-menus 2.4.1-1
    Cinnamon menu specifications
local/cinnamon-screensaver 2.4.2-1
    Screensaver designed to integrate well with the Cinnamon desktop.
local/cinnamon-session 2.4.3-1
    The Cinnamon Session Handler
local/cinnamon-settings-daemon 2.4.3-1
    The Cinnamon Settings daemon
local/cinnamon-translations 2.4.4-1
    Translations for Cinnamon and Nemo
local/cjs 2.4.2-1
    Javascript Bindings for Cinnamon
local/muffin 2.4.5-1
    Cinnamon window manager based on Mutter
local/nemo 2.4.5-1
    Cinnamon file manager (Nautilus fork)`

with geary 0.10.0 on arch linux

**Comment by mtwebster** [comment]
It was for troubleshooting, don't read too far into it.. I am curious if it's starting before muffin, and possibly muffin is causing some (initially) hidden window to become visible once it initializes.

**Comment by ManIVIctorious** [comment]
Okay, i just tried it out with a delay of 60 seconds (startup of cinnamon after logging in is about 20 seconds on my system) and the window still isn't hidden.

**Comment by Vahan86** [comment]
@ManIVIctorious, is this still an issue for you with Cinnamon 3?

**Comment by ManIVIctorious** [comment]
The command `geary --hidden` in startup still doesn't work, I will have to check if this is only a geary specific issue or a problem with all optional flags for programs. I'll check in the next days

**Comment by ManIVIctorious** [comment]
Okay, I just spent two hours with rebooting my computer (took me quite some time to realize that I only had to log out and back in, not very proud of this...) and I found the two causes.

1. @mtwebster was right, some applications need a short delay to work at all (happened with a test bashscript) btw. notify-send seems to take some time too, better write to files in such test cases (took me two reboots...)

2. The issue with geary is, that it starts automatically with the session in hidden mode, when you add a `geary --hidden` to your autostart this activates the user interface. 

So all in all I'll close this issue, but I would really like to know if it is somehow possible to display such "hidden" autostart programs like geary. And if it is possible could this pleas be added to the autostart page in cinnamon-settings?
